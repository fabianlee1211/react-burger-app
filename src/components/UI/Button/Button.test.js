import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('<Button />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Button btnType='Success' disabled clicked={() => {}} />);
  });

  it('renders Button correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Success / Danger type Button', () => {
    wrapper.setProps({ btnType: 'Success' });
    expect(wrapper.find('button.Button.Success')).toHaveLength(1);
    wrapper.setProps({ btnType: 'Danger' });
    expect(wrapper.find('button.Button.Danger')).toHaveLength(1);
  });

  it('renders a disabled button', () => {
    wrapper.setProps({ disabled: true });
    expect(wrapper.find('button').props().disabled).toBe(true);
  });

  it('renders Button text', () => {
    wrapper.setProps({ children: 'Click Here' });
    expect(wrapper.find('button').text()).toBe('Click Here');
  });

  it('can be clicked', () => {
    const clickFn = jest.fn();
    wrapper.setProps({ clicked: clickFn });
    wrapper.find('button').simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
});
