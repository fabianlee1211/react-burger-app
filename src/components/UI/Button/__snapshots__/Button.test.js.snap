// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders Button correctly 1`] = `
<button
  className="Button Success"
  disabled={true}
  onClick={[Function]}
/>
`;
